### ISSUE TITLE
[Feature Request] Add Support for Incognito Mode

### ISSUE THREAD

**Issue by Venderbad** [osfans/trime#938]
**Problem Description**
Trime does not have a built-in incognito mode, which means trime simultaneously record every new word I typed to the user dictionary and record everything I put in the system Pastebin, even if I don't want it to.

**Possible Solution**
So, I think it would be neat to have a built-in Incognito mode which can be manually entered or automatically triggered by the incognito requests from application context, just like GBoard and Microsoft SwiftKey.

**Alternative Solution**
Add multiple profile support and isolate their user dictionary so that we can enter another profile as an incognito mode maybe? But I think it's much more complex than simply add a built-in incognito mode

**Additional context**

References:

[How to Go Incognito on Android Keyboard Using Gboard and SwiftKey](https://www.gtricks.com/android/go-incognito-android-using-gboard-swiftkey/)

**Comment by qwert-f** [comment]
Trime not record user data, the problems you mentioned in the issue about user dictionary, if your schema not automictic record word or sentence, words you input not be recording too. Schema like flypy use closed dictionary not have the problem you asked

**Comment by Venderbad** [comment]
> Schema like flypy use closed dictionary not have the problem you asked

Yes, thanks for your instruction and I think it can solve the problem in time, but not permanently. Using an alternative schema did not solve the core problem. In your way I need to use another schema to replace the one I am currently using, and since that new schema uses a closed dictionary, it will make the IME a completely Incognito keyboard, which is not necessary and could negatively affect daily usage.

Besides, we still cannot auto/manually switch to a global Incognito mode (or switch to another schema) when it should be switched like GBoard or SwiftKey. [This](https://github.com/fcitx5-android/fcitx5-android), however, is providing the Incognito feature. I think if trime could provide the same feature then it would be even more perfect.

**Comment by qwert-f** [comment]
> > Schema like flypy use closed dictionary not have the problem you asked
> 
> Yes, thanks for your instruction and I think it can solve the problem in time, but not permanently. Using an alternative schema did not solve the core problem. In your way I need to use another schema to replace the one I am currently using, and since that new schema uses a closed dictionary, it will make the IME a completely Incognito keyboard, which is not necessary and could negatively affect daily usage.
> 
> Besides, we still cannot auto/manually switch to a global Incognito mode (or switch to another schema) when it should be switched like GBoard or SwiftKey. [This](https://github.com/fcitx5-android/fcitx5-android), however, is providing the Incognito feature. I think if trime could provide the same feature then it would be even more perfect.

Sorry, not good at English, so let you not understand means. I use flypy schema just as a example, not call you use it. Just add a parameters in schema you used can forbid automatic record words you input. you also can add a parameters as a switcher control use or stop.

**Comment by Venderbad** [comment]
> you also can add a parameters as a switcher control use or stop.

If this switcher can be automatically triggered in the right time then this problem is solved. Unfortunately it seems like trime it self does not detect the incognito keyboard request, which means it is not possible to automate this process, so far.

**Comment by WhiredPlanck** [comment]
> > you also can add a parameters as a switcher control use or stop.
> 
> If this switcher can be automatically triggered in the right time then this problem is solved. Unfortunately it seems like trime it self does not detect the incognito keyboard request, which means it is not possible to automate this process, so far.

Trime actually can detect incognito mode of system, but the problem is from rime engine: it doesn't have any runtime switch to turn on/off personel learning, making support of Trime itself for incognito mode is meaningless.
